test/test_utils.jl

```diff
@@ -0,0 +1,390 @@
+using AbstractDifferentiation
+using Test, LinearAlgebra
+using Random
+Random.seed!(1234)
+
+fder(x, y) = exp(y) * x + y * log(x)
+dfderdx(x, y) = exp(y) + y * 1/x
+dfderdy(x, y) = exp(y) * x + log(x)
+
+fgrad(x, y) = prod(x) + sum(y ./ (1:length(y)))
+dfgraddx(x, y) = prod(x)./x
+dfgraddy(x, y) = one(eltype(y)) ./ (1:length(y))
+dfgraddxdx(x, y) = prod(x)./(x*x') - Diagonal(diag(prod(x)./(x*x')))
+dfgraddydy(x, y) = zeros(length(y),length(y))
+
+function fjac(x, y)
+    x + -3*y + [y[2:end];zero(y[end])]/2# Bidiagonal(-ones(length(y)) * 3, ones(length(y) - 1) / 2, :U) * y
+end
+function dfjacdx(x, y)
+    if VERSION < v"1.3"
+        return Matrix{Float64}(I, length(x), length(x))
+    else
+        return I(length(x))
+    end
+end
+dfjacdy(x, y) = Bidiagonal(-ones(length(y)) * 3, ones(length(y) - 1) / 2, :U)
+
+# Jvp
+jxvp(x,y,v) = dfjacdx(x,y)*v
+jyvp(x,y,v) = dfjacdy(x,y)*v
+
+# vJp
+vJxp(x,y,v) = dfjacdx(x,y)'*v
+vJyp(x,y,v) = dfjacdy(x,y)'*v
+
+const xscalar = rand()
+const yscalar = rand()
+
+const xvec = rand(5)
```

Comment:
I wonder if we should move these `const`s somewhere else (maybe in the runtests file?) to avoid the redefinition warning (since the seed is fixed, they can however probably also stay here).

Reply:
Moving them to `runtests.jl` would prevent from running each test file independently. What we could do alternatively is make them non-`const`. We could also hard-code them.

Reply:
@frankschae I opted to move all randomness out of `test_utils.jl`, hardcoding the constants, and allowing the testing functions to accept an RNG. Now each test file is responsible for setting its own seed or providing an RNG.